internal/services/azapi_resource.go

```diff
@@ -499,6 +506,36 @@ func (r *AzapiResource) ModifyPlan(ctx context.Context, request resource.ModifyP
 			return
 		}
 	}
+
+	// Check if any paths in replace_triggers_refs have changed
+	if state != nil && plan != nil && !plan.ReplaceTriggersRefs.IsNull() {
+		refPaths := make(map[string]string)
+		for pathIndex, refPath := range AsStringList(plan.ReplaceTriggersRefs) {
+			refPaths[fmt.Sprintf("%d", pathIndex)] = refPath
+		}
+
+		// read previous values from state
+		var data interface{}
+		err = json.Unmarshal([]byte(state.Body.String()), &data)
+		if err != nil {
+			response.Diagnostics.AddError("Invalid configuration", err.Error())
+			return
+		}
+		previousValues := flattenOutputJMES(data, refPaths)
+
+		// read current values from plan
+		err = json.Unmarshal([]byte(plan.Body.String()), &data)
+		if err != nil {
+			response.Diagnostics.AddError("Invalid configuration", err.Error())
+			return
+		}
+		currentValues := flattenOutputJMES(data, refPaths)
+
+		// compare previous and current values
+		if !reflect.DeepEqual(previousValues, currentValues) {
+			response.RequiresReplace.Append(path.Root("body"))
```

Comment:
Is it possible to add the specific path like "body.properties.sku.name" that triggers the replacement?

Reply:
I tried this, the cli will say:
```
Provider "registry.terraform.io/azure/azapi" has indicated "requires replacement" on azapi_resource.example for a
│ non-existent attribute path cty.Path{cty.GetAttrStep{Name:"body.properties.sku.name"}}.
```

Similar error message if I try with `body.sku.name`